src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/build.py

```diff
@@ -3,288 +3,93 @@
-            consecutive_sleep_in_sec += total_sleep_time
-            logger.debug("Base sleep time: %d, random delay: %d, total: %d, consecutive: %d",
-                         sleep_time, rnd, total_sleep_time, consecutive_sleep_in_sec)
-            time.sleep(total_sleep_time)
-
-    # One final check to see if there's anything in the buffer to flush
-    # E.g., metadata has been set and start == available, but the log file
-    # didn't end in \r\n, so we were unable to flush out the final contents.
-    curr_bytes = stream.getvalue()
-    if curr_bytes:
-        print(curr_bytes.decode('utf-8', errors='ignore'))
-
-    build_status = _get_build_status(metadata).lower()
-    logger.debug("Build status was: '%s'", build_status)
-
-    if raise_error_on_failure:
-        if build_status == 'internalerror' or build_status == 'failed':
-            raise CLIError("Build failed")
-        elif build_status == 'timedout':
-            raise CLIError("Build timed out")
-        elif build_status == 'canceled':
-            raise CLIError("Build was canceled")
-
-
-def _blob_is_not_complete(metadata):
-    if not metadata:
-        return True
-
-    for key in metadata:
-        if key.lower() == 'complete':
-            return False
-
-    return True
-
-
-def _get_build_status(metadata):
-    if metadata is None:
-        return 'inprogress'
-
-    for key in metadata:
-        if key.lower() == 'complete':
-            return metadata[key]
-
-    return 'inprogress'
-
-
-def _get_blob_info(blob_sas_url):
-    match = re.search((r"http(s)?://(?P<account_name>.*?)\.blob\.(?P<endpoint_suffix>.*?)/(?P<container_name>.*?)/"
-                       r"(?P<blob_name>.*?)\?(?P<sas_token>.*)"), blob_sas_url)
-    account_name = match.group('account_name')
-    endpoint_suffix = match.group('endpoint_suffix')
-    container_name = match.group('container_name')
-    blob_name = match.group('blob_name')
-    sas_token = match.group('sas_token')
-
-    if not account_name or not container_name or not blob_name or not sas_token:
-        raise CLIError("Failed to parse the SAS URL: '{!s}'.".format(blob_sas_url))
-
-    return account_name, endpoint_suffix, container_name, blob_name, sas_token
-
-
 def acr_build(cmd,
               client,
               registry_name,
               source_location,
               image_names=None,
               resource_group_name=None,
               timeout=None,
-              build_arg=None,
-              secret_build_arg=None,
-              docker_file_path='Dockerfile',
+              arg=None,
+              secret_arg=None,
+              docker_file_path='',
               no_format=False,
               no_push=False,
               no_logs=False,
-              os_type='Linux'):
+              os_type=OS.linux.value):
     _, resource_group_name = validate_managed_registry(
         cmd.cli_ctx, registry_name, resource_group_name, BUILD_NOT_SUPPORTED)
 
+    from ._client_factory import cf_acr_registries
```

Comment:
Does the new API support both quick build and quick run?

Reply:
No. For `az acr build` we schedule a run and hence using the new api.